handlers/shaqodoon/shaqodoon.go

```diff
@@ -1,6 +1,210 @@
-package shaqdoon
+package shaqodoon
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/courier/utils"
+	"github.com/pkg/errors"
+)
 
 /*
 POST /api/v1/shaqodoon/received/uuid/
 from=252634101111&text=Msg
 */
+
+func init() {
+	courier.RegisterHandler(NewHandler())
+}
+
+type handler struct {
+	handlers.BaseHandler
+}
+
+// NewHandler returns a new External handler
+func NewHandler() courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("SQ"), "External")}
+}
+
+// Initialize is called by the engine once everything is loaded
+func (h *handler) Initialize(s courier.Server) error {
+	h.SetServer(s)
+	s.AddReceiveMsgRoute(h, "POST", "receive", h.ReceiveMessage)
+	s.AddReceiveMsgRoute(h, "GET", "receive", h.ReceiveMessage)
+
+	sentHandler := h.buildStatusHandler("sent")
+	s.AddUpdateStatusRoute(h, "GET", "sent", sentHandler)
+	s.AddUpdateStatusRoute(h, "POST", "sent", sentHandler)
+
+	deliveredHandler := h.buildStatusHandler("delivered")
+	s.AddUpdateStatusRoute(h, "GET", "delivered", deliveredHandler)
+	s.AddUpdateStatusRoute(h, "POST", "delivered", deliveredHandler)
+
+	failedHandler := h.buildStatusHandler("failed")
+	s.AddUpdateStatusRoute(h, "GET", "failed", failedHandler)
+	s.AddUpdateStatusRoute(h, "POST", "failed", failedHandler)
```

Comment:
Can we check the log to see how they actually call us and only remap those methods?